protocols/kad/src/behaviour.rs

```diff
@@ -2492,6 +2547,42 @@ pub enum KademliaEvent {
     PendingRoutablePeer { peer: PeerId, address: Multiaddr },
 }
 
+/// Information about progress events.
+#[derive(Debug, Clone)]
+pub struct ProgressStep {
+    /// The index into the event
+    pub count: NonZeroUsize,
+    /// Is this the final event?
+    pub last: bool,
+}
+
+impl ProgressStep {
+    fn first() -> Self {
+        Self {
+            count: NonZeroUsize::new(1).expect("1 to be greater than 0."),
+            last: false,
+        }
+    }
+
+    fn first_and_last() -> Self {
+        let mut first = ProgressStep::first();
+        first.last = true;
+        first
+    }
+
+    fn next(&self) -> Self {
+        assert!(!self.last);
+
+        Self {
+            count: self
+                .count
+                .checked_add(1)
+                .expect("Not to exhaust usize space."),
```

Comment:
```suggestion
                // TODO: Use checked_add with Rust 1.64.
                .saturating_add(1),
```

This is only in stable with `1.64`.

```
error[E0658]: use of unstable library feature 'nonzero_ops'
    --> protocols/kad/src/behaviour.rs:2579:18
     |
2579 |                 .checked_add(1)
     |                  ^^^^^^^^^^^
     |
     = note: see issue #84186 <https://github.com/rust-lang/rust/issues/84186> for more information
     ```

Reply:
`saturating_add` is only available on `1.64` as well: https://doc.rust-lang.org/std/num/struct.NonZeroUsize.html#method.saturating_add 

Fixed it in an ugly way..

Reply:
> `saturating_add` is only available on `1.64` as well: https://doc.rust-lang.org/std/num/struct.NonZeroUsize.html#method.saturating_add 
> 
> Fixed it in an ugly way.. 

I don't mind bump MSRV as long as it is documented in the changelog :)

Reply:
seems not really worth imho for this